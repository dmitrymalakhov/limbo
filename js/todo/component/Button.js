'use strict';

import React, { Component, PropTypes } from 'react';
import styled from 'styled-components';

import { noop } from '../utils/misc';

const StyleButton = styled.button`
  background: #FAF;
`;

export default class Button extends Component {
  static displayName = 'Button';
  static propTypes = {
    title: PropTypes.string,
    onClick: PropTypes.func,
  };

  static defaultProps = {
    title: '',
    onClick: noop,
  };

  _handleClick = () => {
    this.props.onClick();
  }

  render() {
    return (
      <StyleButton onClick={this._handleClick}>
        {this.props.title}
      </StyleButton>
    );
  }
}
