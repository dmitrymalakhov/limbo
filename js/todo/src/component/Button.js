'use strict';

import React, { Component } from 'react';

class Button extends Component {
	constructor(props) {
		super(props);
		this.displayName = 'Button';
	}

	onClick = () => {
		this.props.onClick && this.props.onClick();
	}

	render() {
		return <button onClick={this.onClick}>{this.props.title}</button>;
	}
}

export default Button;
