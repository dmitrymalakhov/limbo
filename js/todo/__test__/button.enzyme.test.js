import React, { Component } from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from '../component/Button';

describe('<Button/>', () => {
    it('renders correctly with default props', () => {
        const wrapper = mount(
             <Button />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
  }
);
