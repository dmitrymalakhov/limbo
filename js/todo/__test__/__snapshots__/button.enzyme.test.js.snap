exports[`<Button/> renders correctly with default props 1`] = `
<Button
  onClick={[Function]}
  title="">
  <styled.button
    onClick={[Function]}>
    <button
      className="itEXFo"
      onClick={[Function]} />
  </styled.button>
</Button>
`;
